internal/pkg/agent/application/paths/common.go

```diff
@@ -191,7 +196,13 @@ func SetLogs(path string) {
 
 // VersionedHome returns a versioned path based on a TopPath and used commit.
 func VersionedHome(base string) string {
-	return filepath.Join(base, "data", fmt.Sprintf("elastic-agent-%s", release.ShortCommit()))
+	versionedHomePath := filepath.Join(base, "data", fmt.Sprintf("elastic-agent-%s-%s", release.VersionWithSnapshot(), release.ShortCommit()))
```

Comment:
I wish we didn't have to hard code the path again, ideally we'd pull this out of the manifest file, but making that possible needs some more investigation.

This PR should unblock the tests in the larger upgrade change Paolo has, so we can merge this as is and iterate on it from there.